planner/core/task.go

```diff
@@ -435,6 +435,10 @@ func (p *PhysicalHashJoin) GetCost(lCnt, rCnt float64) float64 {
 	probeCost /= float64(p.Concurrency)
 	// Cost of additional concurrent goroutines.
 	cpuCost += probeCost + float64(p.Concurrency+1)*sessVars.ConcurrencyFactor
+	// Cost of traveling the hash table when building the hash table from the outer table
+	if p.OuterHashJoin {
```

Comment:
The `cpuCost` and `memoryCost` in the line 397 and 398 also need to be refined.

Reply:
why? I think the other parts keep the same with others, excepting for here.

Reply:
Because they are calculated using the `innerCnt`, but we use `outerCnt` when `p.OuterHashJoin` is true.

Reply:
```
if (p.InnerChildIdx == 1 && !p.UseOuterToBuild) || (p.InnerChildIdx == 0 && p.UseOuterToBuild) {
	innerCnt, outerCnt = rCnt, lCnt
}
```
the code at Line 393 reversed the inner and the outer, so other parts needn't to be changed.